### PR TITLE
KYC V2 UI upgraded and manage to open both individuals KYC forms

### PR DESCRIPTION
**Pull request by kapildave07**
## Description

The KYC V2 UI has been upgraded, allowing users to open and manage KYC forms for both individual customers(old and new) more efficiently. This enhancement improves the user experience and streamlines the identity verification process.

## Changes

- UI Upgraded: Refers to improvements in the design, functionality, and usability of the interface that users interact with. 
used the same components and code that we are using in other components 
- Open Both Individuals KYC Forms: The upgraded UI now supports accessing and managing KYC forms for different types of individual customers, enhancing efficiency and user experience.

## Attached Links

1. Jira ticket: https://investax.atlassian.net/browse/IXSPD1-1228
https://investax.atlassian.net/browse/IXSPD1-1234
2. Documents:

## Checklist

- [x] Manual tests
- [x] The pull request doesn't affect existing feature

## For reviewers

- Ensure new code doesn't break existing features
- Reviewer shouldn't merge if there are pending unresolved comments

### CODE REVIEW

File: src/pages/AppRoutes.tsx [hunk 1]
```diff
@@ -30,6 +30,7 @@ const SecTokenDetails = lazy(() => import('pages/SecTokenDetails'))
 
 const KYC = lazy(() => import('pages/KYC'))
 const IndividualKYC = lazy(() => import('pages/KYC/IndividualKycForm'))
+const IndividualKYCNew = lazy(() => import('pages/KYC/NewIndividualKycForm'))
```
Comment (thi-investax) [inline src/pages/AppRoutes.tsx:33]: Suggest rename `IndividualKYCV2`

File: src/pages/AppRoutes.tsx [hunk 2]
```diff
@@ -137,6 +138,13 @@ export const routeConfigs: RouteMapEntry[] = [
     component: IndividualKYC,
     conditions: { isWhitelisted: true, kycFormAccess: 'individual' },
   },
+
+  {
+    path: routes.kycIndividualNew,
+    component: IndividualKYCNew,
```
Comment (thi-investax) [inline src/pages/AppRoutes.tsx:144]: I suggest rename `IndividualKycNew` to `IndividualKycV2`

**Review by thuongthanhto1 — APPROVED** [review]

File: src/components/AdminKyc/index.tsx [hunk 3]
```diff
@@ -189,9 +196,14 @@ export const AdminKycTable = () => {
     getKyc()
   }, [id, getKyc])
 
+  console.log(kyc, 'jjjjkjk')
```
Comment (thi-investax) [inline src/components/AdminKyc/index.tsx:199]: Should remove this log

File: src/components/KycReviewModal/index.tsx [hunk 4]
```diff
@@ -19,14 +19,19 @@ import { IndividualForm } from './IndividualForm'
...
   onClose: () => void
   data: KycItem
+  version?: string;
 }
 
-export const KycReviewModal = ({ isOpen, onClose, data }: Props) => {
+export const KycReviewModal = ({ isOpen, onClose, data, version }: Props) => {
+  if (version === 'v2') {
+    return <KycReviewModalV2 isOpen={isOpen} onClose={onClose} data={data} />;
+  }
```
Comment (thi-investax) [inline src/components/KycReviewModal/index.tsx:34]: dont please do conditional rendering outside 

version == 'v2' ? <KycReviewModalV2 isOpen={isOpen} onClose={onClose} data={data} />  : <KycReviewModal>

File: src/components/KycReviewModal/KycReviewModalV2.tsx [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+    if (data?.individualKycId) {
+      return `${data?.individual?.firstName} ${data?.individual?.middleName || ''} ${data?.individual?.lastName || ''}`
+    } else {
+      return data?.corporate?.corporateName || ''
+    }
+  }
+
+  //   if (loadingCynopsis) return <LoadingIndicator isLoading size={96} />
+
+  //   const needResubmit = [KYCStatuses.DRAFT, KYCStatuses.FAILED].includes(data?.status as any)
```
Comment (thi-investax) [inline src/components/KycReviewModal/KycReviewModalV2.tsx:74]: remove all commented code

**Review by thi-investax — CHANGES_REQUESTED** [review]

File: src/components/KycReviewModal/KycReviewModalV2.tsx (hunk 5 above)
Comment (kapildave07) [reply src/components/KycReviewModal/KycReviewModalV2.tsx:74]: some of functions can be useful while APi integrations. That’s why i commented code. after the API integration is done i will remove unused code.

File: src/pages/KYC/NewIndividualKycForm.tsx [hunk 6]
```diff
@@ -137,80 +111,44 @@ export default function NewIndividualKycForm() {
...
+    }
+  }, [])
 
   const handleCheckboxChange = (event: { target: { value: any } }) => {
     const { value } = event.target
     setSelectedCheckbox(value)
   }
+  const checkForErrorsAndEmptyFields = (values: any, errors: any) => {
+    return !values.firstName || !values.lastName || !values.email || errors.firstName || errors.lastName || errors.email
+  }
```
Comment (thi-investax) [inline src/pages/KYC/NewIndividualKycForm.tsx:151]: @kapildave07  I think you are making the same mistake again you made in the LBP. 
this is not the right way to check if a form can be submitted. hi  @thuongthanhto1 ,do you have any recommendation?

**Review by thi-investax — CHANGES_REQUESTED** [review]

File: src/pages/KYC/NewIndividualKycForm.tsx [hunk 7]
```diff
@@ -0,0 +1,480 @@
...
+    form.current.setFieldTouched(key, true)
+  }
+
+  const validationSeen = (key: string) => {
+    if (errors[key]) {
+      const newErrors = { ...errors }
+      delete newErrors[key]
+      setErrors(newErrors)
+    }
+    setCanSubmit(true)
```
Comment (thi-investax) [inline src/pages/KYC/NewIndividualKycForm.tsx:114]: Is it possible that we use errors from Formik.

image

File: src/pages/KYC/EmailVerificationSection.tsx [hunk 8]
```diff
@@ -159,30 +192,41 @@ const CodeInput: React.FC<any> = ({
...
     } catch (error) {
-      handleError('An unexpected error occurred')
+      handleVerificationError('An unexpected error occurred')
       setVerifyError(true)
       setTimer(60)
     }
   }
 
+  const handleVerificationSuccess = () => {
+    if (onSuccess) onSuccess(emailType === 'primary-email' ? 'personal' : 'businessEmail')
```
Comment (thi-investax) [inline src/pages/KYC/EmailVerificationSection.tsx:215]: @kapildave07  could we make these values Enum? later on we just need to update for one place

**Comment by thi-investax** [comment]
image
@kapildave07 , when typing the input show error already. should only show when we don't focus the form

**Comment by thi-investax** [comment]
image
enter a new email but `Email already exist` still appears

File: src/pages/KYC/IndividualKycFormV2.tsx [hunk 9]
```diff
@@ -0,0 +1,495 @@
...
+        {
+          <Formik
+            innerRef={form}
+            initialValues={individualFormV2InitialValues}
+            validationSchema={
+              selectedCheckbox === SecondaryContactTypeV2.BUSINESS_EMAIL
+                ? businessEmailSchema
+                : selectedCheckbox !== SecondaryContactTypeV2.TELEGRAM
+                ? individualErrorsSchemaV2
+                : undefined
```
Comment (thi-investax) [inline src/pages/KYC/IndividualKycFormV2.tsx:214]: this condtion looks unreadable


Please consider refactor into this 

```js
const getValidationSchema = (selectedCheckbox) => {
  if (selectedCheckbox === SecondaryContactTypeV2.BUSINESS_EMAIL) {
    return businessEmailSchema;
  }
  if (selectedCheckbox === SecondaryContactTypeV2.TELEGRAM) {
    return undefined;
  }
  return individualErrorsSchemaV2;
}
```

File: src/pages/KYC/IndividualKycFormV2.tsx [hunk 10]
```diff
@@ -0,0 +1,495 @@
...
+                            placeholder="Email address"
+                            kycVersion={'v2'}
+                            id="emailAddressField"
+                            label="Email address"
+                            value={kyc?.individual?.email || values.email}
+                            error={touched.email && errors.email}
+                            onChange={(e: any) => onChangeInput('email', e.currentTarget.value, values, setFieldValue)}
+                          />
+                        </Column>
+                        {!kyc?.individual?.email && !isPersonalVerified && (
```
Comment (thi-investax) [inline src/pages/KYC/IndividualKycFormV2.tsx:306]: chnage this to !kyc?.individual?.isEmailVerified

File: src/pages/KYC/IndividualKycFormV2.tsx [hunk 11]
```diff
@@ -0,0 +1,495 @@
...
+                              firstName: values.firstName,
+                              middleName: values.middleName,
+                              lastName: values.lastName,
+                              email: values.email,
+                            }}
+                            onSuccess={() => handleSuccess('personal')}
+                          />
+                        )}
+
+                        {isPersonalVerified || kyc?.individual?.email ? <VerificationConfirmation /> : null}
```
Comment (thi-investax) [inline src/pages/KYC/IndividualKycFormV2.tsx:324]: !kyc?.individual?.isEmailVerified
 
 should change isPersonalVerified  -> isPersonalEmailVerified

**Comment by thi-investax** [comment]
should change the component name from `EmailVerificationSection` -> `SecondaryContactOption` .

**Pull request merged**